### ISSUE TITLE
can not select application insights ResourceType in china

### ISSUE THREAD

**Issue by virtualcca** [grafana/grafana#37435]
**What happened**:
can not select application insights resource type
![image](https://user-images.githubusercontent.com/5656078/127817123-5071c5b4-4fd8-48fa-90db-d8f57df3ddc7.png)

but i edit in query inspector like this
![image](https://user-images.githubusercontent.com/5656078/127816948-c5eddff9-59b2-4485-ad1a-b966decc71e8.png)
it can show application insights data 
![image](https://user-images.githubusercontent.com/5656078/127817280-e38cb2b4-f22d-4711-b4e3-d13d18361e5c.png)
 
i am use azure montor in china 

![image](https://user-images.githubusercontent.com/5656078/127817053-d4a48c7a-629a-4a09-ac04-76c9856b07f9.png)

**Environment**:
- Grafana version: 8.0.6
- Data source type & version: Azure Monitor & 3.0.1 (AzureChina)
- OS Grafana is installed on: Windows 2016
- User OS & Browser: Microsoft Edge 92
- Grafana plugins:
- Others:

**Comment by ying-jeanne** [comment]
It seems the frontend is filtering out the application insights resource type, @grafana/cloud-datasources could you please have a look? Thanks

**Comment by joshhunt** [comment]
This issue will be affected by our work here soon for #33710. Once we do that we might ask you to test again, because we are unable to test other national clouds such as Azure China.

**Comment by andresmgot** [comment]
@virtualcca in the Network tab of your browser, when selecting "DEV-CoreArc" in the drop-down, you should see an API call to something like `/subscriptions/<subscription-id>/resourceGroups/<resource-group>/resources?api-version=2018-01-01`. Can you verify if that call return any resource for Application Insights?

If there is nothing related to that in the response, we are not able to detect the resource "InterfaceAPI". In that case, I would contact Azure Monitor support to know why the resource is missing from the response.

Note that you can replicate the same API call directly calling the Azure API. Something like:

```
curl -H "Authorization: Bearer YOUR_TOKEN" https://management.chinacloudapi.cn/subscriptions/<subscription-id>/resourceGroups/<resource-group>/resources?api-version=2018-01-01`
```

Let us know if you are able to find out something else!

**Comment by virtualcca** [comment]
it seem can detect application insights and resource "InterfaceAPI"
![image](https://user-images.githubusercontent.com/5656078/143511503-6a663a1c-8514-4663-b605-16c2a2a67998.png)

but resource type can not select application insights
![image](https://user-images.githubusercontent.com/5656078/143511573-e7458bc3-e0f5-4421-a8b1-c098b2e64d5c.png)

**Comment by andresmgot** [comment]
Thanks for the information @virtualcca ! Turns out that the metric namespace was missing from the supported list. This should be fixed by #42349!
